### PR TITLE
Asynchronously launch script on AWS EC2 instances 

### PR DESCRIPTION
**Pull request by jrochas**

### CODE REVIEW

File: infrastructures/infrastructure-aws-ec2/build.gradle [hunk 1]
```diff
@@ -5,8 +5,8 @@ shadowJar {
 
 }
 
-sourceCompatibility = JavaVersion.VERSION_1_7
-targetCompatibility = JavaVersion.VERSION_1_7
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
```
Comment (Mykhailenko) [inline infrastructures/infrastructure-aws-ec2/build.gradle]: congrats

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AbstractAddonInfrastructure.java [hunk 2]
```diff
@@ -244,29 +233,26 @@ public Void handle() {
...
-                        connectorIaasController.terminateInstance(infrastructureId, instanceId);
-                        logger.info("Instance terminated: " + instanceId);
-                    }
-                    nodesPerInstance.remove(instanceId);
-                    logger.info("Removed instance : " + instanceId);
+        setPersistedInfraVariable(() -> {
+            // first read from the runtime variables map
+            nodesPerInstance = (Map<String, Set<String>>) persistedInfraVariables.get(NODES_PER_INSTANCES_KEY);
+            // make modifications to the nodesPerInstance map
+            nodesPerInstance.get(instanceId).remove(nodeName);
```
Comment (Mykhailenko) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AbstractAddonInfrastructure.java]: maybe it is better to check if `nodeName` existed

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AWSEC2Infrastructure.java#L238](https://github.com/ow2-proactive/nodesource-addons/blob/6beb55132454bd500d768421576776d301f88c2f/infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java#L238): Remove this useless assignment to local variable "instancesIds". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
1. ![MINOR][MINOR] [AWSEC2Infrastructure.java#L59](https://github.com/ow2-proactive/nodesource-addons/blob/6beb55132454bd500d768421576776d301f88c2f/infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java#L59): Rename this field "aws_key" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)
1. ![MINOR][MINOR] [AWSEC2Infrastructure.java#L62](https://github.com/ow2-proactive/nodesource-addons/blob/6beb55132454bd500d768421576776d301f88c2f/infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java#L62): Rename this field "aws_secret_key" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AbstractAddonInfrastructure.java (hunk 2 above)
Comment (jrochas) [reply infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AbstractAddonInfrastructure.java]: Yes I applied. Thanks.

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AWSEC2Infrastructure.java#L238](https://github.com/ow2-proactive/nodesource-addons/blob/b35ab1cbb1f5da29f76da11b206b775c8d6839fb/infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java#L238): Remove this useless assignment to local variable "instancesIds". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
1. ![MINOR][MINOR] [AWSEC2Infrastructure.java#L59](https://github.com/ow2-proactive/nodesource-addons/blob/b35ab1cbb1f5da29f76da11b206b775c8d6839fb/infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java#L59): Rename this field "aws_key" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)
1. ![MINOR][MINOR] [AWSEC2Infrastructure.java#L62](https://github.com/ow2-proactive/nodesource-addons/blob/b35ab1cbb1f5da29f76da11b206b775c8d6839fb/infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java#L62): Rename this field "aws_secret_key" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by jrochas** [comment]
retest this please

**Pull request merged**